metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -314,6 +324,49 @@ class DebugProvider(
     connectedToServer.future.map(_ => server)
   }
 
+  private def startDebugSession(
+      buildServer: BuildServerConnection,
+      params: DebugSessionParams,
+      cancelPromise: Promise[Unit],
+  ) =
+    if (buildServer.isDebuggingProvider || buildServer.isSbt) {
+      buildServer.startDebugSession(params, cancelPromise)
+    } else {
+      def getDebugee: Option[MetalsDebuggee] =
+        params.getDataKind() match {
+          case b.DebugSessionParamsDataKind.SCALA_MAIN_CLASS =>
+            for {
+              id <- params.getTargets().asScala.headOption
+              projectInfo <- debugConfigCreator.create(id)
+              scalaMainClass <- params.asScalaMainClass()
+            } yield new MainClassDebugAdapter(
+              workspace,
+              scalaMainClass,
+              projectInfo,
+              userConfig().javaHome,
+            )
+          case _ => None
+        }
+
+      for {
+        _ <- compilations.compileTargets(params.getTargets().asScala.toSeq)
+      } yield {
+        val debuggee = getDebugee.getOrElse(
+          throw new RuntimeException(s"Can't resolve debugee")
```

Comment:
This will most likely be thrown back to the user, so best to have some along the lines of `Cannot resolve main class` etc. The more information the better.